src/app.js

```diff
@@ -99,24 +99,36 @@ export function app() {
 		 * @param {string} topic Topic.
 		 * @param {*=} data Data.
 		 */
-		emit(topic, target, data = {}) {
-			if (arguments.length < 2) {
+		emit(topic, ...args) {
+			if (args.length === 0 || args.length > 2) {
 				throw new Error(
-					'ts.io().publish() called with invalid arguments.',
+					'ts.io().emit() called with invalid arguments.',
 					arguments
 				);
 			}
-			const message = { type: 'PUBLISH', target, topic, data, token };
+
+			let target, data;
+
+			if (args.length === 1) {
+				target = args[0];
+			} else {
+				data = args[0];
+				target = args[1];
+			}
+
+			const message = { type: 'EVENT', target, topic, data, token };
 			if (token) {
-				debug('%o (%o) to %o - %o', 'PUBLISH', topic, target, data);
+				debug('%o (%o) to %o - %o', 'EVENT', topic, target, data);
 				postMessage(message);
 			} else {
-				debug('%o (%o) to %o - %o', 'PUBLISH(queued)', topic, target, data);
+				debug('%o (%o) to %o - %o', 'EVENT(queued)', topic, target, data);
 				queueMessage(message);
 			}
 		},
-		add(method, handler) {
-			methodHandlers.set(method, handler);
+		define(lifecyle) {
+			if (lifecyle.spawn) {
```

Comment:
`TODO` for the future: Add error messages in case it's passed `lifecycle` events that aren't supported.